lib/process-services-cloud/src/lib/process/process-filters/services/process-filter-cloud.service.ts

```diff
@@ -102,14 +153,20 @@ export class ProcessFilterCloudService {
      * @returns Observable of process instance filters with newly added filter
      */
     addFilter(newFilter: ProcessFilterCloudModel): Observable<ProcessFilterCloudModel[]> {
-        const key: string = this.prepareKey(newFilter.appName);
-        return this.getProcessFiltersByKey(newFilter.appName, key).pipe(
+        const { appName, name } = newFilter;
+        const key: string = this.prepareKey(appName);
+
+        return this.getProcessFiltersByKey(appName, key).pipe(
             switchMap((filters: ProcessFilterCloudModel[]) => {
                 if (filters && filters.length === 0) {
-                    return this.createProcessFilters(newFilter.appName, key, [newFilter]);
+                    return this.createProcessFilters(appName, key, [newFilter]);
                 } else {
+                    const existing = filters.find(filter => filter.name === name);
+                    if (existing) {
+                        filters.splice(filters.indexOf(existing), 1);
+                    }
```

Comment:
Could we use something like this to achieve the same result? the .find and .indexOf will iterate 2 times the filters Array. One to find the element and one to find the index. Instead I think this way is a little bit more efficient. What do you think?
```
const index = filters.findIndex(filter => filter.name === name);
if (index !== -1) {
filters.splice(index, 1);
}
```

Reply:
makes sense, thanks!